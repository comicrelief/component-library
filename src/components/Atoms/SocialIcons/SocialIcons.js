import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import getLinks from './Utils/Links';
import icons from './Utils/Icons';
import Icon from './Icon/Icon';
import spacing from '../../../theme/shared/spacing';

const StyledList = styled.ul`
  display: flex;
  list-style-type: none;
  margin: 0 auto ${spacing('l')};
  justify-content: center;
  padding: 0;
  align-items: center;

  @media ${({ theme }) => theme.breakpoint('small')} {
    margin-bottom: ${spacing('xl')};
    justify-content: start;
  }
`;

const StyledItem = styled.li`
  width: 80px;
  padding: 10px;
`;

/** Social media icons with customizable style linked to campaign social media accounts */
const SocialIcons = ({ campaign, ...restProps }) => {
  const links = getLinks(campaign);

  return (
    <StyledList>
      {Object.keys(icons).map(brand => (
        <StyledItem key={brand}>
          <Icon
            icon={icons[brand]}
            href={links[brand].url}
            title={links[brand].title}
            brand={brand}
            {...restProps}
          />
        </StyledItem>
      ))}
    </StyledList>
  );
};

SocialIcons.propTypes = {
  /** Campaign, used to get social media accounts' links */
  campaign: PropTypes.string.isRequired,
  /** Social media account link target */
  target: PropTypes.string
};

SocialIcons.defaultProps = {
  target: 'blank'
};

export default SocialIcons;
