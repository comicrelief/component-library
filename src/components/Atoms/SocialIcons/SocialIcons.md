### Comic Relief

```js
<SocialIcons campaign="comicrelief" />
```

### Red Nose Day

```js
<SocialIcons campaign="rednoseday" />
```

### Sport Relief with self target

```js
<SocialIcons campaign="sportrelief" target="self" />
```
