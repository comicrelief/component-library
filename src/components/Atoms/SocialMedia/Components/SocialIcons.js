import React from 'react';
import PropTypes from 'prop-types';
import StyledSocialIcon from './SocialIcon.style';

const SocialIcons = ({ className, icons, links, ...restProps }) => {
  return (
    <ul className={`icons ${className}`}>
      {Object.keys(icons).map(brand => (
        <li key={brand}>
          <StyledSocialIcon
            icon={icons[brand]}
            href={links[brand]}
            {...restProps}
          />
        </li>
      ))}
    </ul>
  );
};

SocialIcons.propTypes = {
  className: PropTypes.string.isRequired,
  links: PropTypes.shape({
    facebook: PropTypes.string,
    twitter: PropTypes.string,
    youtube: PropTypes.string,
    instagram: PropTypes.string
  }).isRequired,
  icons: PropTypes.shape({
    facebook: PropTypes.object,
    twitter: PropTypes.object,
    youtube: PropTypes.object,
    instagram: PropTypes.object
  }).isRequired
};

export default SocialIcons;
