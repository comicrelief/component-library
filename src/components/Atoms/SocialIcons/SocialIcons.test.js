import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import SocialIcons from './SocialIcons';

it('renders correctly with Comic Relief links', () => {
  const tree = renderWithTheme(<SocialIcons campaign="comicrelief" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Sport Relief links', () => {
  const tree = renderWithTheme(<SocialIcons campaign="sportrelief" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with Red Nose Day links and self target', () => {
  const tree = renderWithTheme(
    <SocialIcons campaign="rednoseday" target="self" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
