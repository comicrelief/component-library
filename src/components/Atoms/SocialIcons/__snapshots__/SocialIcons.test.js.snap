// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Comic Relief links 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  display: block;
  position: relative;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
}

.c2:hover,
.c2:focus {
  opacity: 0.6;
}

.c3 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 32px;
  margin-right: 1.5rem;
}

@media (min-width:740px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
  }
}

@media (min-width:740px) {
  .c1 {
    width: 48px;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-facebook"
      href="https://www.facebook.com/comicrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Facebook page"
    >
      <img
        alt="facebook"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-instagram"
      href="https://www.instagram.com/comicrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Instagram account"
    >
      <img
        alt="instagram"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-twitter"
      href="https://twitter.com/comicrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Twitter account"
    >
      <img
        alt="twitter"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-youtube"
      href="https://www.youtube.com/channel/UCdF5u0ggeSETozc8fsprjcw"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our YouTube channel"
    >
      <img
        alt="youtube"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
</ul>
`;

exports[`renders correctly with Red Nose Day links and self target 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  display: block;
  position: relative;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
}

.c2:hover,
.c2:focus {
  opacity: 0.6;
}

.c3 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 32px;
  margin-right: 1.5rem;
}

@media (min-width:740px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
  }
}

@media (min-width:740px) {
  .c1 {
    width: 48px;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-facebook"
      href="https://www.facebook.com/rednoseday"
      rel="noopener noreferrer"
      target="_self"
      title="Check out our Facebook page"
    >
      <img
        alt="facebook"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-instagram"
      href="https://www.instagram.com/rednoseday"
      rel="noopener noreferrer"
      target="_self"
      title="Check out our Instagram account"
    >
      <img
        alt="instagram"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-twitter"
      href="https://twitter.com/rednoseday"
      rel="noopener noreferrer"
      target="_self"
      title="Check out our Twitter account"
    >
      <img
        alt="twitter"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-youtube"
      href="https://www.youtube.com/channel/UCdF5u0ggeSETozc8fsprjcw"
      rel="noopener noreferrer"
      target="_self"
      title="Check out our YouTube channel"
    >
      <img
        alt="youtube"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
</ul>
`;

exports[`renders correctly with Sport Relief links 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  display: block;
  position: relative;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
}

.c2:hover,
.c2:focus {
  opacity: 0.6;
}

.c3 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 32px;
  margin-right: 1.5rem;
}

@media (min-width:740px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
  }
}

@media (min-width:740px) {
  .c1 {
    width: 48px;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-facebook"
      href="https://www.facebook.com/sportrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Facebook page"
    >
      <img
        alt="facebook"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-instagram"
      href="https://www.instagram.com/sportrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Instagram account"
    >
      <img
        alt="instagram"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-twitter"
      href="https://twitter.com/sportrelief"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our Twitter account"
    >
      <img
        alt="twitter"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      data-testid="icon-youtube"
      href="https://www.youtube.com/channel/UCdF5u0ggeSETozc8fsprjcw"
      rel="noopener noreferrer"
      target="_blank"
      title="Check out our YouTube channel"
    >
      <img
        alt="youtube"
        className="c3"
        src="mock.asset"
      />
    </a>
  </li>
</ul>
`;
